app/models/manageiq/providers/ibm_cloud/provider.rb

```diff
@@ -0,0 +1,72 @@
+class ManageIQ::Providers::IbmCloud::Provider < ::Provider
+  has_many :power_virtual_servers_cloud_managers,
+           :foreign_key => "provider_id",
+           :class_name  => "ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager",
+           :inverse_of  => :provider,
+           :dependent   => :destroy
+
+  def self.params_for_create
+    {
+      :fields => [
+        {
+          :component   => "select",
+          :id          => "provider_id",
+          :name        => "provider_id",
+          :label       => _("IBM Cloud Provider"),
+          :isClearable => true,
+          :options     => Rbac.filtered(ManageIQ::Providers::IbmCloud::Provider.all).pluck(:name, :id).map do |name, id|
+            {
+              :label => name,
+              :value => id.to_s,
+            }
+          end
+        },
+        {
+          :component  => "password-field",
+          :name       => "authentications.default.auth_key",
+          :id         => "authentications.default.auth_key",
+          :label      => _("IBM Cloud API Key (if not using an existing provider)"),
+          :type       => "password",
+          :isRequired => true,
+          :condition  => {
+            :when    => "provider_id",
+            :isEmpty => true
+          },
```

Comment:
Hey @skateman this condition worked great but I realized we can't edit the authentication with this hidden after creation then.

Can we change the "required" to be contingent on if the provider is selected or not?

Reply:
You can add the field twice with mutually exclusive conditions and having/not having the validation...

Reply:
Hm this didn't work, they both show up and I get an error about duplicate keys

Reply:
```
                      {
                        :component  => "password-field",
                        :name       => "authentications.default.auth_key",
                        :id         => "authentications.default.auth_key",
                        :label      => _("IBM Cloud API Key (Required)"),
                        :type       => "password",
                        :isRequired => true,
                        :condition  => {
                          :when    => "provider_id",
                          :isEmpty => true
                        },
                        :validate   => [{:type => "required"}]
                      },
                      {
                        :component  => "password-field",
                        :name       => "authentications.default.auth_key",
                        :id         => "authentications.default.auth_key-optional",
                        :label      => _("IBM Cloud API Key (Optional)"),
                        :type       => "password",
                        :isRequired => false,
                        :condition  => {
                          :when    => "provider_id",
                          :isEmpty => false
                        }
                      }
```

Reply:
Yeah, we might need a custom validator, that only gets triggered if a condition is set. I'm not sure what should I do with the `*` marking yet but I'll figure that out too.

Reply:
Okay for now I'll just leave the authentications.default.auth_key required

Reply:
So it seems you can't have the field with the same name twice :confused: and we cannot do `resolveProps` as we send the schema as pure JSON. I will make a decorator around all the validators and enhance it by allowing a condition in the validator's body. This way the validation would only run if the condition is met. I actually had this as a proposed feature in the upstream DDF but it has been removed from v2 :disappointed: 

The `isRequired` is a little bit trickier, I have a few options, but all of them seem crazily hacky for now. I will keep thinking while I'm implementing the conditional validators. In the meantime I have to ask :trollface: is it necessary to hide the `*` from the field?

Reply:
https://github.com/data-driven-forms/react-forms/issues/802